src/cli/actions/hub/login.js

```diff
@@ -16,41 +16,47 @@ async function pollTokenUrl (tokenUrl, deviceCode, interval) {
   logger.http(`POST ${tokenUrl} with deviceCode ${deviceCode} at interval ${interval}`)
 
   try {
-    const response = await axios.post(tokenUrl, {
-      client_id: OAUTH_CLIENT_ID,
-      device_code: deviceCode,
-      grant_type: 'urn:ietf:params:oauth:grant-type:device_code'
+    const response = await request(tokenUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        client_id: OAUTH_CLIENT_ID,
+        device_code: deviceCode,
+        grant_type: 'urn:ietf:params:oauth:grant-type:device_code'
+      })
     })
 
-    logger.http(response.data)
+    const responseData = await response.body.json()
 
-    if (response.data.access_token) {
-      spinner.start()
-      store.setToken(response.data.full_username, response.data.access_token)
-      store.setHostname(response.data.hostname)
-      spinner.succeed(`logged in as ${response.data.username}`)
-      process.exit(0)
-    } else {
-      // continue polling if no access_token. shouldn't ever get here it server is implemented correctly
-      setTimeout(() => pollTokenUrl(tokenUrl, deviceCode, interval), interval * 1000)
-    }
-  } catch (error) {
-    if (error.response && error.response.data) {
-      logger.http(error.response.data)
+    logger.http(responseData)
 
+    if (response.statusCode !== 200) {
```

Comment:
i will change this to `>= 400` to handle future possibility someone strict about codes changes the api's response code to 201.